src/modules/extra/m_ssl_openssl.cpp

```diff
@@ -704,13 +704,23 @@ class OpenSSLIOHook : public SSLIOHook
 		certinfo->activation = GetTime(X509_getm_notBefore(cert));
 		certinfo->expiration = GetTime(X509_getm_notAfter(cert));
 
-		int activated = ASN1_UTCTIME_cmp_time_t(X509_getm_notBefore(cert), ServerInstance->Time());
-		if (activated != -1 && activated != 0)
-			certinfo->error = "Certificate not activated";
+		int activated = ASN1_TIME_cmp_time_t(X509_getm_notBefore(cert), ServerInstance->Time());
+		if (activated != -1 && activated != 0) {
+			char date[64];
+			struct tm tm;
+			gmtime_r(&certinfo->activation, &tm);
+			strftime(date, 64, "%b %e %T %Y %Z", &tm);
+			certinfo->error = "Certificate not activated (notBefore '" + std::string(date) + "')";
```

Comment:
This is obviously hacky C-style code and almost certainly inappropriate for C++, but I wasn't going to try and battle search engines to figure out which of the 500 string-concatenation methods I should actually use.  If you let me know what method to use then I can refactor accordingly.

The output was made to mirror that given by `openssl x509 -in foo.pem -text`.  Example from my fork with some enhanced OpenSSL logging:
```
            Sun May 07 2023 21:42:01 m_ssl_openssl: Invalid peer certificate chain from '127.0.0.1' port '6697' ('Certificate has expired (notAfter 'Dec 25 20:25:54 2120 GMT'), Cert chain #1: Certificate has expired (notAfter 'Jun  9 07:12:20 2120 GMT'), Cert chain #2: Certificate has expired (notAfter 'Jun  9 07:12:17 2120 GMT')'):
```